pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -61,6 +62,27 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		cr.Spec.Version = ""
 	}
 
+	// if the secret is the only credential present
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminUser) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
+		creds := constants.KIE_ADMIN_CREDENTIALS_SECRET
+		if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 {
+			creds = cr.Spec.CommonConfig.SecretAdminCredentials
+		}
+		errSecret, _, _ := checkAndCreateAdminSecret(service, creds, cr.Namespace, cr.Spec.Environment)
+		if errSecret != nil {
+			log.Error("Can't create Admin Secret. ", errSecret)
+		}
+	}
+
+	//in case of missing secretAdmin we use the AdminUsername AdminPassword and if missing we set the defaults
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
+		password := constants.DefaultPassword
+		if !isTrial(cr) {
+			password = string(shared.GeneratePassword(8))
+		}
+		cr.Spec.CommonConfig.AdminPassword = password
+	}
```

Comment:
is this part really needed? I think this should be already as part of operator. Isn't it covered earlier in code or elsewhere?

Reply:
is needed

Reply:
can't you reuse that func that already sets the password?